server/models/UserModel/index.js

```diff
@@ -0,0 +1,48 @@
+import config from '../../configs/config'
+import { collections } from '../../utils/mongoCollections'
+import { USER_MONGO_PROJECTION } from '../../constants'
+
+const UserModel = {
+  withDefaults: (overrides = {}) => ({
+    display_name: '',
+    title: '',
+    department: '',
+    company: '',
+    mail: '',
+    member_of: '',
+    bad_pwd_count: 0,
+    lockout_time: 0,
+    last_logoff: Date.now(),
+    last_logon: Date.now(),
+    account_expires: null,
+    password: '',
+    ldap: false,
+    force_reset_pw: false,
+    realms: config.app.realms,
+    icon: '',
+    access: [],
+    blocked: false,
+    role: 'member',
+    ...overrides,
+  }),
+  save: async (db, userAttributes) => {
+    const newUser = UserModel.withDefaults(userAttributes)
+
+    return await db.collection(collections.users).insertOne(newUser)
+  },
+  findUserAndUpdate: async (db, userQuery, userUpdates) => {
+    return await db.collection(collections.users).findOneAndUpdate(
+      userQuery,
```

Comment:
what do you think about passing in the `uid` instead of `userQuery` here, and then constructing the query in the function? I think it would be a nice interface for the caller.

```js
const user = await UserModel.find(db, uid, { last_logon: Date.now() })
```

Reply:
[25454a1](https://github.com/AMP-SCZ/dpdash/pull/293/commits/25454a1384e02684707ecd6ca148e3dc2ad4032f)